core/src/test/java/org/infinispan/security/CacheAuthorizationTest.java

```diff
@@ -91,6 +95,49 @@ public Void run() {
       });
    }
 
+   public void testAdminCombinations() throws Exception {
+      List<Supplier<Void>> calls = Arrays.asList(() -> {
+         cacheManager.getGlobalComponentRegistry();
+         return null;
+      }, () -> {
+         cacheManager.getCacheManagerConfiguration();
+         return null;
+      });
+
+      for (final AuthorizationPermission perm : AuthorizationPermission.values()) {
+         for (Supplier<Void> fn : calls) {
+            Supplier<Void> lazySubjectCall = () -> {
+               try {
+                  Security.doAs(SUBJECTS.get(perm), (PrivilegedExceptionAction<Void>) () -> {
+                     fn.get();
+                     return null;
+                  });
+               } catch (Exception e) {
+                  throw new IllegalStateException(e);
```

Comment:
I don't think there is a need for the `lazySubjectCall` supplier here, you can just create a `PrivilegedExceptionAction` variable and pass it to `Security.doAs`. You can also catch the `PrilegedActionException` directly:

```java
            PrivilegedExceptionAction<Void> action = () -> {
               // perform some action
               return null;
            };
            if (AuthorizationPermission.ADMIN.equals(perm)) {
               Security.doAs(SUBJECTS.get(perm), action);
            } else {
               try {
                  Security.doAs(SUBJECTS.get(perm), action);
               } catch (PrivilegedActionException exceptionAction) {
                   // add test logic
               }
            }
```